### PR TITLE
[BugFix] Fix unknown error when create materialized view without bucket num

### PR DESCRIPTION
**Pull request by Youngwb**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes https://github.com/StarRocks/StarRocksTest/issues/1121

## Problem Summary(Required) ：

Fix two bugs in mv
1.  In this PR https://github.com/StarRocks/starrocks/pull/10614, SR set default bucket num from 10 to 0, this would result in create mv failed when user not set bucket num. according to that PR, we use be numner to calculate  bucket num
2.  SR do not support create mv on view, so we need to disable it at analyze stage

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by Youngwb** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 16 / 16 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzerUtils.java|8|8|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|5|5|100.00%|[]|

**Review by murphyatwork — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java [hunk 1]
```diff
@@ -74,6 +77,11 @@ public enum RefreshTimeUnit {
             SECOND
         }
 
+        private boolean isSupportBasedOnTable(Table table) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java:80]: What's the method name meaning?

**Pull request merged**